lib/mongoose_push/service/apns/error_handler.ex

```diff
@@ -0,0 +1,53 @@
+defmodule MongoosePush.Service.APNS.ErrorHandler do
+  @moduledoc """
+  Module responsible for handling errors returned by APNS service
+  """
+  alias MongoosePush.Service
+
+  # More information about possible return codes is here:
+  # https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html
+  @spec translate_error_reason(Service.error_reason()) :: Service.error()
+  def translate_error_reason(:BadCollapseId), do: {:invalid_request, :BadCollapseId}
+  def translate_error_reason(:BadDeviceToken), do: {:invalid_request, :BadDeviceToken}
+  def translate_error_reason(:BadExpirationDate), do: {:invalid_request, :BadExpirationDate}
+
+  def translate_error_reason(:DeviceTokenNotForTopic),
+    do: {:invalid_request, :DeviceTokenNotForTopic}
```

Comment:
Let's rename `unregistered` to `invalid_device_token` and group all errors that relate to invalid device token.